swagger_server/controllers/connection_controller.py

```diff
@@ -89,14 +89,14 @@ def place_connection(body):
     db_instance.add_key_value_pair_to_db("connection_data", json.dumps(body))
     logger.info("Saving to database complete.")
 
-    topo_val = db_instance.read_from_db("latest_topo")["latest_topo"]
-    topo_json = json.loads(topo_val)
-
-    logger.info(f"Read topology {topo_val}")
+    if db_instance.read_from_db("latest_topo"):
+        topo_val = db_instance.read_from_db("latest_topo")["latest_topo"]
+        topo_json = json.loads(topo_val)
```

Comment:
Few thoughts:

- It is better to change `read_from_db()` such that we can get what we need by simply calling `read_from_db("latest_topo")`.
- We'll get an error if `db_instance.read_from_db("latest_topo")` returns `None`, or if `"latest_topo"` key does not exist in the dict returned.
- We'll get another error from `json.loads()` if `topo_val` turns out to be `None`.

Reply:
Yes exactly. Was thinking to update this as well.